leetcode/314_binary_tree_vertical_order_traversal.py

```diff
@@ -0,0 +1,53 @@
+from typing import List, Optional
+from collections import defaultdict, deque
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
+        """Represents a node in a binary tree.
+        Args:
+            val (int, optional): node value. Defaults to 0.
+            left (_type_, optional): left child. Defaults to None.
+            right (_type_, optional): right child. Defaults to None.
+        """
+        self.val = val
+        self.left = left
+        self.right = right
+
+class Solution:
+    def verticalOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
+        """Returns the vertical order traversal of a binary tree.
+
+        Args:
+            root (Optional[TreeNode]): The root of the binary tree.
+
+        Returns:
+            List[List[int]]: A list of lists containing the nodes' values in vertical order.
+        """
+        if not root:
+            return []
+
+        column_items = defaultdict(list)
+        queue = deque([(0,root)])
+
+        res = []
+
+        leftmost_col = float('inf')
+        rightmost_col = float('-inf')
+
+        while queue:
+            col, node = queue.popleft()
+
+            column_items.append(node.val)
+            leftmost_col = min(leftmost_col, col)
+            rightmost_col = max(rightmost_col, col)
+
+            if node.left:
```

Comment:
Usually checks like these should be done at the current node level, not the children. E.g. you should have 

```
if node is None:
     continue
```

This way it's safe to append `None` to your queue. because the previous check will handle it. 

Imagine if you had an n-ary tree with more than 2 children, then you'd have a lot more if/else blocks like this. Checking at the current node makes this easier.